File: reference-code/src/main/java/devsetup/Shading.java
package devsetup;

import dev.jorel.commandapi.CommandAPI;
import dev.jorel.commandapi.CommandAPIBukkitConfig;
import dev.jorel.commandapi.CommandAPICommand;
import org.bukkit.plugin.java.JavaPlugin;

class Shading {
    {
        JavaPlugin plugin = null;
        // #region bukkitConfigExample
        CommandAPI.onLoad(new CommandAPIBukkitConfig(plugin).silentLogs(true));
        // #endregion bukkitConfigExample
    }

    // #region shadingExample
    class MyPlugin extends JavaPlugin {
        @Override
        public void onLoad() {
            CommandAPI.onLoad(new CommandAPIBukkitConfig(this).verboseOutput(true)); // Load with verbose output

            new CommandAPICommand("ping")
                .executes((sender, args) -> {
                    sender.sendMessage("pong!");
                })
                .register();
        }

        @Override
        public void onEnable() {
            CommandAPI.onEnable();
            // Register commands, listeners etc.
        }

        @Override
        public void onDisable() {
            CommandAPI.onDisable();
        }
    }
    // #endregion shadingExample
}

File: reference-code/src/main/kotlin/devsetup/Shading.kt
package devsetup

import dev.jorel.commandapi.CommandAPI
import dev.jorel.commandapi.CommandAPIBukkitConfig
import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.executors.NormalExecutor
import org.bukkit.command.CommandSender
import org.bukkit.plugin.java.JavaPlugin

fun shading() {
    val plugin: JavaPlugin? = null
    // #region bukkitConfigExample
    CommandAPI.onLoad(CommandAPIBukkitConfig(plugin).silentLogs(true))
    // #endregion bukkitConfigExample
}

// #region shadingExample
class MyPlugin : JavaPlugin() {
    override fun onLoad() {
        CommandAPI.onLoad(CommandAPIBukkitConfig(this).verboseOutput(true)) // Load with verbose output

        CommandAPICommand("ping")
            .executes(NormalExecutor<CommandSender, Any> { sender, _ ->
                sender.sendMessage("pong!")
            })
            .register()
    }

    override fun onEnable() {
        CommandAPI.onEnable()
        // Register commands, listeners etc.
    }

    override fun onDisable() {
        CommandAPI.onDisable()
    }
}
// #endregion shadingExample